File: App_Be_Spring Boot/src/main/java/com/cashier/app_spring/model/Transactions.java
package com.cashier.app_spring.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Table(name = "transactions", schema = "public")
public class Transactions {

    @Id
    @Column(name = "id")
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "qr_code")
    private Customers customer;

    @ManyToOne
    @JoinColumn(name = "rf_id")
    private Product product;

    @Column(name = "price")
    private Integer price;

    @Column(name = "quantity")
    private Integer quantity;

    @Column(name = "total_price")
    private Integer totalPrice;

    @Column(name = "date")
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime date;
}

File: App_Be_Express/src/model/transactions.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    qrCode: { type: String, required: true },
    rfId: { type: String, required: true },
    price: { type: Number, required: true },
    quantity: {type:Number, required: true},
    totalPrice: { type: Number, required: true },
    date: { type: String, required: true },
});

module.exports = mongoose.model('Transaction', transactionSchema);
